front_end/test/downloads.js

```diff
@@ -0,0 +1,187 @@
+            const fileName = "2mib.txt";
+            const bucketName = "us-west1";
+            const result = await downloads.timeDownload(fileName, bucketName);
+            assert.equal(result, 0);
+        });
+
+        it('should build URL correctly', async () => {
+            const fileName = "2mib.txt";
+            const bucketName = "us-west1";
+            await downloads.timeDownload(fileName, bucketName);
+            assert.equal(downloads._builtURL, `https://storage.googleapis.com/gcsrbpa-us-west1/2mib.txt`);
+        });
+    });
+
+    describe('benchmarkSingleDownload', () => {
+        it('should return a Map object on success', async () => {
+            const fileName = "2mib.txt";
+            const bucketName = "us-west1";
+            const result = await downloads.benchmarkSingleDownload(fileName, bucketName);
+
+            let expected = new Map();
+            expected.set('bucketName', 'us-west1');
+            expected.set('location', 'Oregon');
+            expected.set('fileName', '2mib.txt');
+            expected.set('timeTaken', 0);
+            expected.set('fileSizeBytes', 2097152);
+            expected.set('speedBps', 'Infinity');
+            expected.set('speedMiBps', 'Infinity');
+
+            assert.deepEqual(result, expected);
+        });
+
+        it('should throw Error if fileName is invalid', async () => {
+            const fileName = "random_file_name";
+            const bucketName = "us-west1";
+
+            assert.rejects(downloads.benchmarkSingleDownload(fileName, bucketName), "Invalid File Name");
+        });
+
+        it('should throw Error if bucketName is invalid', async () => {
+            const fileName = "2mib.txt";
+            const bucketName = "random_bucket_name";
+
+            assert.rejects(downloads.benchmarkSingleDownload(fileName, bucketName), "Invalid Bucket Name");
+        });
+
+        it('should return Map with bad values if GET request fails', async () => {
+            const fileName = "wrongFile";
+            const bucketName = "wrongBucket";
+
+            let result = await downloads.benchmarkSingleDownload(fileName, bucketName);
+
+            let expected = new Map();
+            expected.set('bucketName', 'wrongBucket');
+            expected.set('fileName', 'wrongFile');
+            expected.set('fileSizeBytes', 'wrongFile');
+            expected.set('location', 'wrongBucket');
+            expected.set('speedBps', '-1.000');
+            expected.set('speedMiBps', '-1.000');
+            expected.set('timeTaken', -0.001);
+
+            assert.deepEqual(result, expected);
+        })
+    });
+
+    describe('benchmarkDownload', () => {
+        it('should throw Error if fileName is invalid', async () => {
+            const fileName = "random_file_name";
+            const bucketName = "us-west1";
+
+            assert.rejects(downloads.benchmarkDownload(fileName, bucketName), "Invalid File Name");
```

Comment:
Another thing the linter should be catching but you are mixing single quotes and double quotes in this file. I think we generally prefer single.